### ISSUE TITLE
Do you update BERT parameters?

### ISSUE THREAD

**Issue by zhuang-li** [stanford-oval/genienlp#182]
Hi I see you set the learning rate as "--transformer_lr_multiply 0.1". Shouldn't that be too large for BERT? Do you update BERT parameters or just set the parameters fixed?

**Comment by gcampax** [comment]
We do update the bert params when training. I don't remember what hparams we used in the most recent bert models. I suggest looking at the pages associated with each paper, listed from https://wiki.almond.stanford.edu/releases

Recently, we have been using Bart as the base model, and for that we use 0.01 LR multiplier

**Comment by s-jse** [comment]
I'll also add that, as the name suggests, `--transformer_lr_multiply` is a multiplier. So the actual learning rate will be this, divided by the square root of the hidden layers' dimension (=768 for BERT base) [here](https://github.com/stanford-oval/genienlp/blob/master/genienlp/train.py#L646). This formula is what the [original transformer paper](https://arxiv.org/abs/1706.03762) used for the learning rate.

This way, for example using `--transformer_lr_multiply=0.1` gives a learning rate of 0.003 for BERT base. You can compare this number with other repositories that use BERT, like HuggingFace's transformers. Our learning rate is still higher than theirs, because our `TransformerLSTM` model has an LSTM decoder that has not been pre-trained before.

**Comment by nrser** [comment]
Hope we have answered your questions, feel free to engage further on our [forum](https://community.almond.stanford.edu)
